packages/amplify-schema-validator/src/validators/index-scalar-types.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  DocumentNode,
+  Kind,
+  ObjectTypeDefinitionNode,
+  FieldDefinitionNode,
+  EnumTypeDefinitionNode,
+} from 'graphql';
+import { InvalidDirectiveError } from '../exceptions/invalid-directive-error';
+import { isScalarOrEnum } from '../helpers/is-scalar-or-enum';
+import { getTypeDefinitionsOfKind } from '../helpers/get-type-definitions-of-kind';
+
+/**
+ * Validates that every @index directive is of a scalar type
+ *
+ * @param schema graphql schema
+ * @returns true
+ */
+export const validateIndexScalarTypes = (schema: DocumentNode): Error[] => {
+  const errors: Error[] = [];
+  const objectTypeDefinitions = schema.definitions.filter(
```

Comment:
Index can be defined only on 'model' types. Can we add that validation here?
Also consider filtering like (type is OBJECT_TYPE_DEFINITION and contains a 'model' directive).

Reply:
Ideally these validators would be single-purpose to keep each validation rule as simple and discrete as possible.

Perhaps this should be it's own future validator?